app/controllers/api/admin/commodities/search_references_controller.rb

```diff
@@ -46,12 +46,16 @@ def subheading
           end
         end
 
+        def admin_commodity_id
+          params[:admin_commodity_id].presence || params[:commodity_id]
+        end
+
         def commodity_id
-          params[:commodity_id].split('-', 2).first
+          admin_commodity_id.split('-', 2).first
```

Comment:
I had to rename the resource in the separated engine to avoid collisions on path helpers, here. It'll make no difference to the caller of this api